ext/spl/spl_directory.c

```diff
@@ -2046,8 +2046,11 @@ static int spl_filesystem_file_read(spl_filesystem_object *intern, int silent) /
 		intern->u.file.current_line_len = 0;
 	} else {
 		if (SPL_HAS_FLAG(intern->flags, SPL_FILE_OBJECT_DROP_NEW_LINE)) {
-			line_len = strcspn(buf, "\r\n");
-			buf[line_len] = '\0';
+			const char *eol = memchr(buf, '\n', line_len);
+			if (eol != NULL) {
+				line_len = (eol > buf && *(eol - 1) == '\r') ? eol - buf - 1 : eol - buf;
+				buf[line_len] = '\0';
+			}
```

Comment:
As we only need to deal with a newline at the end, I would write it like this:

```suggestion
			if (line_len > 0 && buf[line_len - 1] == '\n') {
			    line_len--;
			    if (line_len > 0 && buf[line_len - 1] == '\r') {
			        line_len--;
			    }
			    buf[line_len] = '\0';
			}
```

Reply:
Ah, right. Thank you!